docs/expectation_gallery/azure-pipelines-expectation-gallery.yml

```diff
@@ -75,23 +75,116 @@ stages:
                 [ScriptsChanged]
                 assets/scripts/**
 
-  - stage: deploy_gallery_staging
+  - stage: exp_tests_on_all_backends
     dependsOn: scope_check
     condition: or(eq(variables.isScheduled, true), eq(variables.isRelease, true), eq(variables.isManual, true))
     pool:
       vmImage: 'ubuntu-20.04'
 
     jobs:
-      - job: build_gallery
-        timeoutInMinutes: 420
+      - job: dataframes
+        timeoutInMinutes: 180
+        variables:
+          python.version: '3.8'
+          GE_pytest_opts: ''
+
+        steps:
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '$(python.version)'
+            displayName: 'Use Python $(python.version)'
+
+          - bash: python -m pip install --upgrade pip
+            displayName: 'Update pip'
+
+          - script: |
+              pip install \
+                "cryptography==38.0.4" \
+                --requirement reqs/requirements-dev-all-contrib-expectations.txt \
+                --editable ".[spark]" \
+                --constraint constraints-dev.txt
+            displayName: 'Install dependencies'
+
+          - bash: python ./build_gallery.py --backends "spark, pandas" 2>&1 | tee output--build_gallery--dataframes.txt ; grep -o "Took .* seconds to .*" output--build_gallery--dataframes.txt | sort -k2,2nr > testing-times--dataframes.txt ; grep -o "ERROR - (.*" output--build_gallery--dataframes.txt | sort > testing-error-messages--dataframes.txt; touch gallery-tracebacks.txt; grep -o "Expectation type.*" output--build_gallery--dataframes.txt | sort > gallery-exp-types--dataframes.txt ; mv checklists.txt checklists--dataframes.txt ; mv gallery-tracebacks.txt gallery-tracebacks--dataframes.txt
```

Comment:
Is there a benefit to doing multiple backends in a single job? Could we parametrize these instead and run each backend separately with the output file using the name of the backend? I think that would be clearer when looking at the output and probably faster as well since it can be parallelized.

Reply:
You would also be able to parametrize the diagnostics so they are not repeated multiple times in this file.

Reply:
@anthonyburdi decided not to parametrize since the differences between the stages are significant enough. But all backends are in their own stages now.

Reply:
I disagree, but this is something we can address later.

Reply:
Some require services and extra scripts, some extra environment variables. Not sure it's worth the effort https://learn.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops&tabs=script